### PR TITLE
feat: added spotify oauth2 provider

### PR DESCRIPTION
**Pull request by Intevel**
Seperated in new PR

### CODE REVIEW

File: docs/content/en/providers/spotify.md
Comment (bmulholland) [inline docs/content/en/providers/spotify.md:35]: Can you please add a section to the bottom of the README under "Development" with instructions on how to get Spotify working in local development?

**Comment by bmulholland** [comment]
Please also add a "Login with Spotify" button to the demo site and ensure that's tested and working

File: src/utils/provider.ts
Comment (bmulholland) [inline src/utils/provider.ts]: This is leaking Spotify specifics out of the Spotify provider and into the general provider code. We need to separate those concerns. How can this be uncoupled such that the main provider class can be configured (or use a param or whatever) to do the same thing, with Spotify using that configuration option?

**Comment by Intevel** [comment]
There is a button ^^

**Comment by bmulholland** [comment]
Oh haha -- thanks! :)

File: docs/content/en/providers/spotify.md
Comment (Intevel) [reply docs/content/en/providers/spotify.md:35]: I placed it at the top, so you can see it directly. The usage is the same.

File: src/utils/provider.ts
Comment (michelmueller) [reply src/utils/provider.ts]: While developing a PR for this I found out, that passing the `client-secret` in the payload of the post-request is also accepted. 
[Official Spotify documentation](https://developer.spotify.com/documentation/general/guides/authorization/code-flow/) states, that the `client-secret` and the `client-id` needs to be passed in the Authorization header but the alternative works as well.

However the [OAuth 2.0 spec](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1) states 

> Including the client credentials in the request-body using the two parameters is NOT RECOMMENDED and SHOULD be limited to clients unable to directly utilize the HTTP Basic authentication scheme (or other password-based HTTP authentication schemes). 

But in the current implementation the `client-secret` is passed either way in the payload of the post-request.

What do you think @bmulholland @Intevel ?

Comment (bmulholland) [reply src/utils/provider.ts]: @michelmueller Yeah, that's a good point. Will be tricky to implement without breaking people's setup. It's also more general than Spotify, so I think it's worth splitting into a separate issue (though may end up being a dependency of this PR?)

Comment (michelmueller) [reply src/utils/provider.ts]: I suggest, that the spotify section in the general provider code in this PR gets removed. So the spotify provider works and does not violate the OAuth 2.0 specification. Honestly I've never contributed to a project yet. So hopefully it is not rude to ask you to do that @Intevel ? :-)

@bmulholland  Agree. But we could add another configuration option to the `Oauth2SchemeOptions` which defaults to the current behavior. When you change the behavior in the config, the `client-secret` gets moved from the body to the HTTP Authorization header. This should be no breaking change.

Comment (Intevel) [reply src/utils/provider.ts]: As it looks this should not be a problem, I'll comment it out for now. What solution do we have for this problem instead? 
And of course it is no problem to ask @michelmueller :-)

File: src/utils/provider.ts
Comment (michelmueller) [inline src/utils/provider.ts]: @Intevel  This change is obsolete now, am I correct? There is no need for any changes in the `provider.ts`

Comment (Intevel) [reply src/utils/provider.ts]: Ah, you're right. I overlooked it, I'll edit it!

File: src/utils/provider.ts
Comment (michelmueller) [reply src/utils/provider.ts]: I've drafted a possible solution in this PR (https://github.com/nuxt-community/auth-module/pull/1656)

File: docs/content/en/providers/spotify.md
Comment (bmulholland) [reply docs/content/en/providers/spotify.md:35]: I don't see any changes to the README?

Comment (bmulholland) [reply docs/content/en/providers/spotify.md:35]: (Spoiler: I'm going to follow the directions you write in the README to set up and test this before merging :) )

Comment (Intevel) [reply docs/content/en/providers/spotify.md:35]: Yeah sure.

File: docs/content/en/providers/spotify.md
Comment (bmulholland) [inline docs/content/en/providers/spotify.md:10]: Is this still the current status? What needs to be complete before we take off this warning?

**Comment by bmulholland** [comment]
Thanks for requesting my review! That alerts me that it's ready to review in a respectful way -- please continue to use that for anything else you think is ready to go.

**Review by bmulholland — APPROVED** [review]
Approved subject to the question above. Will merge when that's resolved.

File: docs/content/en/providers/spotify.md
Comment (Intevel) [reply docs/content/en/providers/spotify.md:10]: No it should be ready for production. I will remove this warning.

**Comment by Intevel** [comment]
But you have to merge this to: https://github.com/nuxt-community/auth-module/pull/1656

**Comment by bmulholland** [comment]
#1656 is marked as draft. Is this PR dependent on #1656?

**Comment by Intevel** [comment]
Yes

**Comment by bmulholland** [comment]
The last comment on that PR says "When https://github.com/nuxt-community/auth-module/pull/1483 is merged, I will update this PR accordingly and test it again."

There appears to be a circular dependency, then. Can you please work out what order these should be merged in? And, if #1656 needs to be merged first, then it will need to be in a shippable state (and non-draft status) before I can merge it.

**Comment by Intevel** [comment]
I'm drafting this for now, I need to take another look at the two PR's.

**Comment by bmulholland** [comment]
Thanks for working on all that :). Dependencies like this can be annoying.

**Pull request closed without merging**
